src/app.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * If you would like to turn your application into a standalone executable, look at server.js file
+ */
+
+import path from "node:path";
+import fastifyAutoload from "@fastify/autoload";
+import { FastifyInstance, FastifyPluginOptions } from "fastify";
+
+export default async function serviceApp(
+  fastify: FastifyInstance,
+  opts: FastifyPluginOptions
+) {
+  // This loads all external plugins defined in plugins/external
+  // those should be registred first as your custom plugins might depend on them
+  fastify.register(fastifyAutoload, {
+    dir: path.join(import.meta.dirname, "plugins/external"),
+    options: { ...opts }
+  });
+
+  // This loads all your custom plugins defined in plugins/custom
+  // those should be support plugins that are reused
+  // through your application
+  fastify.register(fastifyAutoload, {
+    dir: path.join(import.meta.dirname, "plugins/custom"),
+    options: { ...opts }
+  });
```

Comment:
I've suggested this way of doing things, but it remains my intuition.

Also, I wonder if I should add a `priority` export to autoloader, to sort the tree by plugin priority?

```ts

export const autoConfig = { }
export const pluginPriority = 1
export default env;
```

Reply:
Although this is easily achievable with file naming.